docker/fedora-39.dockerfile

```diff
@@ -22,25 +27,63 @@ ENV BUILD_VERSION=${BUILD_VERSION}
 ENV COMMIT=${COMMIT}
 
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
-# install dependencies
+# install build dependencies
 # hadolint ignore=DL3041
 RUN <<_DEPS
 #!/bin/bash
 set -e
+if [[ "${TARGETPLATFORM}" == "linux/amd64" ]]; then
+  export TARGETARCH=x86_64
+elif [[ "${TARGETPLATFORM}" == "linux/arm64" ]]; then
+  export TARGETARCH=aarch64
+else
+  echo "unsupported platform: ${TARGETPLATFORM}"
+  exit 1
+fi
```

Comment:
Or for tidiness…

```bash
case "${TARGETPLATFORM}" in
  linux/amd64) export TARGETARCH=x86_64 ;;
  linux/arm64) export TARGETARCH=aarch64 ;;
  *) echo "unsupported platform: ${TARGETPLATFORM}"; exit 1 ;;
esac
```

Or to solve the other issue at the same time…

```bash
case "${TARGETPLATFORM}" in
  linux/amd64) echo TARGETARCH=x86_64 >> env ;;
  linux/arm64) echo TARGETARCH=aarch64 >> env ;;
  *) echo "unsupported platform: ${TARGETPLATFORM}"; exit 1 ;;
esac

source env
```

Reply:
```bash
case "${TARGETPLATFORM}" in
  linux/amd64) echo TARGETARCH=x86_64 >> env ;;
  linux/arm64) echo TARGETARCH=aarch64 >> env ;;
  *) echo "unsupported platform: ${TARGETPLATFORM}"; exit 1 ;;
esac

source env
```

This option will allow the env to be used in other steps?

Reply:
You just need to call `source env` in the other steps.